project/src/components/offer-card/offer-card.tsx

```diff
@@ -1,53 +1,76 @@
+import {useHistory} from 'react-router-dom';
+import {useDispatch, useSelector} from 'react-redux';
+import classNames from 'classnames';
+
 import type {Offer} from '../../types/offer';
 
 import CommentsSection from '../comments-section/comments-section';
 
-import {getRatingPercentValue, formatType} from '../../utils/utils';
+import {formatType, getRatingPercentValue} from '../../utils/utils';
+import {AppRoute, AuthorizationStatus, CardTypes, FavoriteStatus} from '../../const';
+
+import {getAuthorizationStatus} from '../../store/app-user/selectors';
+
+import {fetchFavoriteOffers, postFavorite} from '../../store/actions/api-actions/api-actions-favorite';
 
 
 type OfferCardProps = {
   card: Offer;
+  typeCard: CardTypes;
 };
 
 
-function OfferCard({card}: OfferCardProps): JSX.Element {
-  const {
-    title,
-    rating,
-    type,
-    bedrooms,
-    maxAdults,
-    price,
-    goods,
-    host,
-    description,
-    isPremium,
-    isFavorite,
-    images,
-  } = card;
+function OfferCard({card, typeCard}: OfferCardProps): JSX.Element {
+  const {title, rating, type, bedrooms, maxAdults, price, goods, host, description, isPremium, isFavorite, images, id} = card;
+
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const authorizationStatus = useSelector(getAuthorizationStatus);
 
   const ratingPercentValue = getRatingPercentValue(rating);
-  const typeText = formatType(type);
+  const typeValue = formatType(type);
+
+  const bedroomsValue = (bedrooms > 1) ? `${bedrooms} Bedrooms` : `${bedrooms} Bedroom`;
+  const priceValue = `€${price}`;
+  const maxAdultsValue = (maxAdults > 1) ? `Max ${maxAdults} adults`: `Max ${maxAdults} adult`;
 
   const checkedImagesOverflow = (images.length > 5) ? images.slice(0, 6) : images;
 
+  const isImages = checkedImagesOverflow.length > 0;
+  const isGoods = goods.length > 0;
+  const isAuth = authorizationStatus === AuthorizationStatus.Auth;
+
+
+  function handleFavoriteClick() {
+    if (isAuth) {
+      const status = isFavorite ? FavoriteStatus.RemoveFromFavorite: FavoriteStatus.AddToFavorite;
+      dispatch(postFavorite({id, status, typeCard}));
+      dispatch(fetchFavoriteOffers(authorizationStatus));
+    } else {
+      history.push(AppRoute.Login);
+    }
+  }
+
   return(
     <>
       <div className="property__gallery-container container">
         <div className="property__gallery">
-          {
-            (checkedImagesOverflow.length)
-              ? checkedImagesOverflow.map((image) => <div key={image + new Date().getTime()} className="property__image-wrapper"><img className="property__image" src={image} alt={title}/></div>)
-              : ''
-          }
+          {isImages && checkedImagesOverflow.map((image) => <div key={image} className="property__image-wrapper"><img className="property__image" src={image} alt={title}/></div>)}
```

Comment:
сложно читать, лучше разнести на разные строки
```suggestion
          {isImages && checkedImagesOverflow.map((image) => (
            <div key={image} className="property__image-wrapper">
              <img className="property__image" src={image} alt={title}/>
            </div>)
          )}
```